src/librc/librc.c

```diff
@@ -596,6 +596,24 @@ rc_is_user(void)
 	return is_user;
 }
 
+int scriptdirfds[SCRIPTDIR_MAX];
+
+int rc_scriptdirfd(const char * const *scriptdir) {
+	const char * const *scriptdir_array = is_user ? rc_dirs.scriptdirs : scriptdirs;
+	size_t idx;
+
+	if (!scriptdir || !*scriptdir || scriptdir < scriptdir_array)
+		return -1;
+
+	if ((idx = scriptdir - scriptdir_array) > SCRIPTDIR_MAX)
```

Comment:
Same issue with pointer arithmetic. Doing arithmetic with unrelated pointer is UB: https://www.iso-9899.info/n3220.html#6.5.7p10

> When two pointers are subtracted, both shall point to elements of the same array object [...]

And so if this condition is true, you're in UB land already. Compiler can straight up remove this check entirely.

(Also, it probably meant to be `>=`).